Sources/PackageCollections/Providers/GitHubPackageMetadataProvider.swift

```diff
@@ -147,7 +147,14 @@ struct GitHubPackageMetadataProvider: PackageMetadataProvider {
 
     internal func apiURL(_ url: String) -> Foundation.URL? {
         do {
-            let regex = try NSRegularExpression(pattern: #"([^/@]+)[:/]([^:/]+)/([^/]+)\.git$"#, options: .caseInsensitive)
+            var url = url.lowercased()
+            
+            let gitSuffix = ".git"
+            if url.hasSuffix(gitSuffix) {
+                url.removeLast(gitSuffix.count)
+            }
+            
+            let regex = try NSRegularExpression(pattern: #"([^/@]+)[:/]([^:/]+)/([^/]+)$"#, options: .caseInsensitive)
```

Comment:
Could we control for this by adjusting the original regex pattern? This would have the advantage of preserving the case of the original URL. 

```suggestion
             let regex = try NSRegularExpression(pattern: #"([^/@]+)[:/]([^:/]+)/([^/]+)(\.git)?$"#, options: .caseInsensitive)
```

Reply:
Extending the original regex makes sense to me.

Reply:
I believe I tried that (and others) before but it didn't work as expected (hence this bug). I will try again though.

Reply:
OK. Figured it out. Looks like the trick is to include `.` in the negative chars class also or else `.git` always gets captured with the repo name group.